services/121-service/src/payments/transactions/repositories/transaction.view.scoped.repository.ts

```diff
@@ -182,37 +181,28 @@ export class TransactionViewScopedRepository extends ScopedRepository<Transactio
       fspName: Fsps;
     }[]
   > {
-    const chunkSize = 2000;
-    const results: {
-      transactionId: number;
-      transferValue: number;
-      referenceId: string;
-      fspName: Fsps;
-    }[] = [];
-
-    for (const idsChunk of chunk(transactionIds, chunkSize)) {
-      const chunkResult = await this.createQueryBuilder('transaction')
-        .innerJoinAndSelect('transaction.registration', 'registration')
-        .innerJoinAndSelect('registration.programFspConfiguration', 'fspConfig')
-        .andWhere('transaction.id IN (:...transactionIds)', {
-          transactionIds: idsChunk,
-        })
-        .select([
-          'transaction.id as "transactionId"',
-          'transaction.transferValue as "transferValue"',
-          'registration."referenceId" as "referenceId"',
-          '"fspConfig"."fspName" as "fspName"',
-        ])
-        .getRawMany<{
-          transactionId: number;
-          transferValue: number;
-          referenceId: string;
-          fspName: Fsps;
-        }>();
-      results.push(...chunkResult);
+    if (transactionIds.length === 0) {
+      return [];
     }
 
-    return results;
+    return await this.createQueryBuilder('transaction')
+      .innerJoinAndSelect('transaction.registration', 'registration')
+      .innerJoinAndSelect('registration.programFspConfiguration', 'fspConfig')
+      .andWhere('transaction.id = ANY(:transactionIds)', {
```

Comment:
so this ANY instead of IN is just a magic-trick to solve too-big-query issues without any consequences? Why does IN still exist then?

Reply:
Most performance increase came from this PR to main https://github.com/global-121/121-platform/pull/7437, which I rebased back in the feature branch

Reply:
Yup, it's a postgres specific magic trick @jannisvisser